src/Native/System.Security.Cryptography.Native/pal_x509_name.cpp

```diff
@@ -22,3 +22,76 @@ extern "C" X509_NAME* DecodeX509Name(const unsigned char* buf, int32_t len)
 
     return d2i_X509_NAME(nullptr, &buf, len);
 }
+
+extern "C" void X509NameDestroy(X509_NAME* a)
+{
+    if (a != nullptr)
+    {
+        X509_NAME_free(a);
+    }
+}
+
+static int32_t X509NamePrint(BIO* out, X509_NAME* nm, unsigned long flags)
+{
+    return X509_NAME_print_ex(out, nm, /*indent*/ 0, flags);
+}
+
+extern "C" int32_t X509NamePrintForFind(BIO* out, X509_NAME* nm)
+{
+    const unsigned long findFormat = XN_FLAG_FN_NONE | XN_FLAG_SEP_CPLUS_SPC;
+
+    return X509NamePrint(out, nm, findFormat);
+}
+
+/*
+Converts from X500DistinguishedNameFlags to the correct flags used in X509_NAME_print_ex.
+*/
+static unsigned long ConvertFormatFlags(X500DistinguishedNameFlags inFlags)
+{
+    unsigned long outFlags = 0;
+
+    if ((inFlags & X500DistinguishedNameFlags::Reversed) != 0)
+    {
+        outFlags |= XN_FLAG_DN_REV;
+    }
+
+    if ((inFlags & X500DistinguishedNameFlags::UseSemicolons) != 0)
+    {
+        outFlags |= XN_FLAG_SEP_SPLUS_SPC;
+    }
+    else if ((inFlags & X500DistinguishedNameFlags::UseNewLines) != 0)
+    {
+        outFlags |= XN_FLAG_SEP_MULTILINE;
+    }
+    else
+    {
+        outFlags |= XN_FLAG_SEP_CPLUS_SPC;
+    }
+
+    if ((inFlags & X500DistinguishedNameFlags::DoNotUseQuotes) != 0)
+    {
+        // TODO: Handle this.
```

Comment:
It looks like @bartonjs took care of these TODOs with https://github.com/dotnet/corefx/pull/3899.  I'm going to leave them in for now, and when he merges with my change he will remove them.